geoh5py/io/h5_writer.py

```diff
@@ -434,12 +457,76 @@ def write_color_map(
             entity_type_handle = H5Writer.fetch_handle(h5file, entity_type)
             map_values = color_map.values
             entity_type_handle.create_dataset(
-                "Color map", map_values.shape, data=map_values, dtype=map_values.dtype
+                "Color map",
+                map_values.shape,
+                data=map_values,
+                dtype=map_values.dtype,
+                compression="gzip",
+                compression_opts=9,
+            )
+
+        if close_file:
+            h5file.close()
+
+    @classmethod
+    def write_value_map(
+        cls,
+        entity_type: "shared.EntityType",
```

Comment:
**question (non-blocking):** since we write value map, should not the entity by typed more specifically?

Reply:
Sorry, i m not getting you here? Like passing the value_map itself?

Reply:
entity is the object data here right? Should the type be `ReferenceData`?

Reply:
so the `value_map` is stored under the `entity_type`. We could pass the `ReferenceData` and get its `entity_type`. Right now I m just passing the nearest level which the data is under. Up to you.

Reply:
ho, right the mapping it is stored on the type. So it could take `DataType` instead of `EntityType`